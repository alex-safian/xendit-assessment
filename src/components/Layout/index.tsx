import React from "react";
import cx from "classnames";
import Header from "@components/Header";

type Props = {
  className?: string;
  children: React.ReactNode;
};

const Layout: React.FC<Props> = (props) => {
  const { children, className } = props;

  return (
    <div className={cx(className)}>
      <Header />
      <div className="pt-32">{children}</div>
    </div>
  );
};

export default Layout;
