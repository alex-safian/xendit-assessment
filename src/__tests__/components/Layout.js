import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Layout from "@components/Layout";
import { BrowserRouter as Router } from "react-router-dom";

it("renders", () => {
  const { asFragment } = render(
    <Router>
      <Layout>content</Layout>
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
