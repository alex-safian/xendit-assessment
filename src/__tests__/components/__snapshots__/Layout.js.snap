// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <header
      class="fixed top-0 right-0 left-0 bg-white shadow z-20"
    >
      <div
        class="container"
      >
        <nav
          class="flex items-end py-6"
        >
          <a
            class="mr-auto md:mr-28"
            href="/"
          >
            <svg
              class="w-auto h-11 lg:h-14"
            >
              xendit.svg
            </svg>
          </a>
          <div
            class="hidden md:flex flex-1 desktop"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              href="/search"
            >
              Search
            </a>
            <a
              class="nav-link"
              href="/newsletter"
            >
              Newsletter
            </a>
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
            <a
              class="ml-auto self-center"
              href="https://github.com/alex-safian/zendit-assessment"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
            </a>
          </div>
          <div
            class="md:hidden"
          >
            <div>
              <div
                class="bm-overlay"
                style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
              />
              <div>
                <div
                  class="bm-burger-button"
                  style="z-index: 1000;"
                >
                  <button
                    id="react-burger-menu-btn"
                    style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                  >
                    Open Menu
                  </button>
                  <span>
                    <span
                      class="bm-burger-bars"
                      style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
                    />
                    <span
                      class="bm-burger-bars"
                      style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
                    />
                    <span
                      class="bm-burger-bars"
                      style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
                    />
                  </span>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="bm-menu-wrap mobile-menu"
                id=""
                style="position: fixed; right: 0px; z-index: 1100; width: 300px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
              >
                <div
                  class="bm-menu"
                  style="height: 100%; box-sizing: border-box; overflow: auto;"
                >
                  <nav
                    class="bm-item-list"
                    style="height: 100%;"
                  >
                    <a
                      aria-current="page"
                      class="nav-link active"
                      href="/"
                    >
                      Home
                    </a>
                    <a
                      class="nav-link"
                      href="/search"
                    >
                      Search
                    </a>
                    <a
                      class="nav-link"
                      href="/newsletter"
                    >
                      Newsletter
                    </a>
                    <a
                      class="nav-link"
                      href="/about"
                    >
                      About
                    </a>
                    <a
                      class="ml-auto self-center"
                      href="https://github.com/alex-safian/zendit-assessment"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <svg
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                        />
                      </svg>
                    </a>
                  </nav>
                </div>
                <div>
                  <div
                    class="bm-cross-button"
                    style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
                  >
                    <button
                      id="react-burger-cross-btn"
                      style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                      tabindex="-1"
                    >
                      Close Menu
                    </button>
                    <span
                      style="position: absolute; top: 6px; right: 14px;"
                    >
                      <span
                        class="bm-cross"
                        style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                      />
                      <span
                        class="bm-cross"
                        style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </header>
    <div
      class="pt-32"
    >
      content
    </div>
  </div>
</DocumentFragment>
`;
